cookbook/offscreen-clipboard-write/offscreen.js

```diff
@@ -17,34 +17,46 @@
 // workflows where secondary uses for the document (or alternate offscreen documents) might be
 // implemented.
 
+// Registering this listener when the script is first executed ensures that the offscreen document
+// will be able to receive messages when the promise returned by `offscreen.createDocument()`
+// resolves.
 chrome.runtime.onMessage.addListener(handleMessages);
 
+// This function performs basic filtering and error checking on messages before dispatching the
+// message to a more specific message handler.
 async function handleMessages(message) {
   // Return early if this message isn't meant for the offscreen document.
   if (message.target !== 'offscreen-doc') {
     return;
   }
 
-  // Return early if we got the wrong type of message.
-  if (message.type === 'copy-data-to-clipboard') {
-    handleClipboardWrite(message.data);
+  // Dispatch the message to an appropriate handler.
+  switch (message.type) {
+    case 'copy-data-to-clipboard':
+      handleClipboardWrite(message.data);
+      break;
+    default:
+      console.warn(`Unexpected message type received: '${message.type}'.`);
   }
 }
 
-// The `handleClipboardWrite()` function makes use of a web platform API (`document`) that is not
-// exposed to extension service workers.
-
+// Use the offscreen document's `document` interface to write a new value ot the system clipboard
 async function handleClipboardWrite(data) {
-  // Return early if we received the wrong kind of data.
+  // Error if we received the wrong kind of data.
   if (typeof data !== 'string') {
     throw new TypeError(`Value provided must be a 'string', got '${typeof data}'.`);
     return;
   }
 
   // Write data the clipboard.
+
+  // `document.execCommand('copy')` works against the user's selected in a web page. As such, we
+  // need to insert the updated text into the document and select it before it can be added to the
+  // clipboard.
   textEl.value = data;
   textEl.select();
   document.execCommand('copy');
 }
 
+// This textarea is used to easily modify and select it's text value.
```

Comment:
The point of filing https://github.com/GoogleChrome/chrome-extensions-samples/issues/795 was to explore another way to solve https://github.com/GoogleChrome/chrome-extensions-samples/issues/766 by writing a `File` to the clipboard, and get the `File` on arbitrary Web pages - not to paste plain text to the clipboard.

Reply:
Resolving this conversation as it's a duplicate of the discussion in [this comment thread](https://github.com/GoogleChrome/chrome-extensions-samples/pull/797#discussion_r1063973635).